agent/tests/test_skills_executor.py

```
@@ -1689,3 +2031,149 @@ def test_display_has_both_options(self, mock_rate, MockId, MockClient,
+
+    def test_no_wallet_required(self):
+        """public_balance should NOT call require_wallet."""
+        with patch("iconfucius.config.require_wallet") as mock_rw:
+            # Will fail on canister calls, but that's fine — we just
+            # check that require_wallet is never called
+            try:
+                execute_tool("public_balance", {
+                    "principal": "aamu4-v26cr-pc5oe-35oal-b3tju-srmh5-cpqql-rkqzf-umm4n-vlahk-hqe",
+                })
+            except Exception:
+                pass
```

Ruff (0.15.2)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)